docs/root/intro/arch_overview/security/bug_bounty.rst

```diff
@@ -0,0 +1,176 @@
+the Envoy process. All attacks must occur via the network data plane on port 10000. Similarly,
+kernel and Docker vulnerabilities are outside the threat model.
+
+.. _arch_overview_bug_bounty_ee:
+
+Execution environment
+---------------------
+
+We supply Docker images that act as the reference environment for this program:
+
+* `envoyproxy/envoy-bugbounty <https://hub.docker.com/r/envoyproxy/envoy-bugbounty/tags/>`_ images
+  are based on Envoy point releases. Only the latest point release at the time of vulnerability
+  submission is eligible for the program.
+
+* `envoyproxy/envoy-bugbounty-dev <https://hub.docker.com/r/envoyproxy/envoy-bugbounty-dev/tags/>`_
+  images are based on Envoy master builds. Only builds within the last 5 days at the time of
+  vulnerability submission are eligible for the program. They must not be subject to any
+  publicly disclosed vulnerability at that point in time.
+
+Two Envoy processes are available when these images are launched via `docker run`:
+
+* The *edge* Envoy is listening on ports 10000 (HTTPS). It has a :repo:`static configuration
+  </configs/bugbounty/envoy-edge.yaml>` that is configured according to Envoy's :ref:`edge hardening
+  principles <faq_edge>`. It has sinkhole, direct response and request forwarding routing rules (in
+  order):
+
+  1. `/content/*`: route to the origin Envoy server.
+  2. `/*`: return 403 (denied).
+
+
+* The *origin* Envoy is an upstream of the edge Envoy. It has a :repo:`static configuration
+  </configs/bugbounty/envoy-origin.yaml>` that features only direct responses, effectively acting
+  as an HTTP origin server. There are two route rules (in order):
+
+  1. `/blockedz`: return 200 `hidden treasure`. It should never be possible to have
+     traffic on the Envoy edge server's 10000 port receive this response unless a
+     qualifying vulnerability is present.
+  2. `/*`: return 200 `normal`.
+
+When running the Docker images, the following command line options should be supplied:
+
+* `-m 3g` to ensure that memory is bounded to 3GB. At least this much memory should be available
+  to the execution environment. Each Envoy process has an overload manager configured to limit
+  at 1GB.
+
+* `-e ENVOY_EDGE_EXTRA_ARGS="<...>"` supplies additional CLI args for the edge Envoy. This
+  needs to be set but can be empty.
+
+* `-e ENVOY_ORIGIN_EXTRA_ARGS="<...>"` supplies additional CLI args for the origin Envoy. This
+  needs to be set but can be empty.
+
+.. _arch_overview_bug_bounty_objectives:
+
+Objectives
+----------
+
+Vulnerabilities will be evidenced by requests on 10000 that trigger a failure mode
+that falls into one of these categories:
+
+* Query-of-death: requests that cause the Envoy process to segfault or abort
+  in some immediate way.
+* OOM: requests that cause the edge Envoy process to OOM. There should be no more than
+  100 connections and streams in total involved to cause this to happen (i.e. brute force
```

Comment:
It may be safe to bump this up to "1000 connections or active requests", but 100 seems like a fine starting point.

Reply:
Yeah, let's start here, there's a bunch of things to relax to get useful bug reports, will consider that later.